### PR TITLE
std.datetime_doc_update 0 of 9

### PR DESCRIPTION
**Pull request by burner**

### CODE REVIEW

File: std/datetime.d [hunk 1]
```diff
@@ -17605,6 +17599,12 @@ assert(Interval!Date(Date(1996, 1, 2), dur!"years"(3)) ==
             throw new DateTimeException("Arguments would result in an invalid Interval.");
     }
 
+    ///
+    unittest
+    {
+        assert(Interval!Date(Date(1996, 1, 2), dur!"weeks"(3)) ==
+               Interval!Date(Date(1996, 1, 2), Date(1996, 1, 23)));
```
Comment (quickfur) [inline std/datetime.d]: Nice catch

**Comment by quickfur** [comment]
LGTM, thanks for taking the time to break this up into more manageable chunks!

**Comment by quickfur** [comment]
Auto-merge toggled on

**Comment by quickfur** [comment]
Auto-merge toggled off

File: std/datetime.d [hunk 2]
```diff
@@ -17632,18 +17632,18 @@ assert(Interval!Date(Date(1996, 1, 2), dur!"years"(3)) ==
...
     @property TP begin() const pure nothrow
     {
         return cast(TP)_begin;
     }
 
+    ///
+    unittest
+    {
+        assert(Interval!Date(Date(1996, 1, 2), Date(2012, 3, 1)).begin ==
+            Date(1996, 1, 2));
```
Comment (quickfur) [inline std/datetime.d]: Nitpick: I think we should try to stick to the original formatting (the wrapped line should line up with the opening `(`).

File: std/datetime.d [hunk 3]
```diff
@@ -17665,18 +17665,18 @@ assert(Interval!Date(Date(1996, 1, 2), Date(2012, 3, 1)).begin ==
...
     @property TP end() const pure nothrow
     {
         return cast(TP)_end;
     }
 
+    ///
+    unittest
+    {
+        assert(Interval!Date(Date(1996, 1, 2), Date(2012, 3, 1)).end ==
+            Date(2012, 3, 1));
```
Comment (quickfur) [inline std/datetime.d]: ditto

File: std/datetime.d [hunk 4]
```diff
@@ -17698,33 +17698,34 @@ assert(Interval!Date(Date(1996, 1, 2), Date(2012, 3, 1)).end ==
...
     @property auto length() const pure nothrow
     {
         return _end - _begin;
     }
 
+    ///
+    unittest
+    {
+        assert(Interval!Date(Date(1996, 1, 2), Date(2012, 3, 1)).length ==
+        dur!"days"(5903));
```
Comment (quickfur) [inline std/datetime.d]: ditto

**Comment by quickfur** [comment]
Hmph, what is going on with the autotester on macmini? It seems to be failing intermittently for reasons unrelated to PRs.

**Comment by schveiguy** [comment]
> Hmph, what is going on with the autotester on macmini? It seems to be failing intermittently for reasons unrelated to PRs.

I think this could be another race condition. Since we turned on the debug test for druntime, tests that depend on exclusive resources (such as an output file that doesn't live in the build directory) are subject to racing. There was a -cov issue that I looked at earlier, looks eerily similar.

**Comment by schveiguy** [comment]
And certain build hosts are more prone to it, due to the cpu speed and number of cores.

File: std/datetime.d (hunk 1 above)
Comment (burner) [inline std/datetime.d]: lets say thanks to the assert

File: std/datetime.d (hunk 2 above)
Comment (burner) [inline std/datetime.d]: If you look at the and of the PR, not all old example follow the same formatting.

I hope I made them all equal

File: std/datetime.d (hunk 1 above)
Comment (quickfur) [inline std/datetime.d]: Yet another reason *all* code examples need to be unittests rather than unmaintained code in ddoc comments.

**Comment by quickfur** [comment]
Fair enough.

**Comment by quickfur** [comment]
Auto-merge toggled on

**Comment by burner** [comment]
thanks

**Comment by schveiguy** [comment]
@quickfur This should fix the random failures: https://github.com/D-Programming-Language/druntime/pull/1390

Please review.

**Pull request merged**